tcs-service/src/main/java/com/transformuk/hee/tis/tcs/service/service/impl/PostServiceImpl.java

```diff
@@ -664,7 +666,11 @@ public PostDTO findOne(Long id) {
   @Override
   public void delete(Long id) {
     log.debug("Request to delete Post : {}", id);
-    postRepository.deleteById(id);
+    if(permissionService.canDeletePostData()) {
+      postRepository.deleteById(id);
+      List<PostFunding> postFundingsToDelete = postFundingRepository.findByPostId(id);
+      postFundingRepository.deleteAll(postFundingsToDelete);
+    }
```

Comment:
If the users don't have the permission `post:delete:entities`, they won't get any error.

Reply:
I have moved it to hasAuthority